packages/node-app/src/utils/inside.util.ts

```diff
@@ -24,20 +26,27 @@ const isValidLocation = (currentLocation, requiredLocation) => {
   return distance < requiredLocation.radius;
 };
 
-const validateLocation = (participant, giveaway) => {
+const validateLocation = async (participant, giveaway) => {
   const participantLocation = participant.location;
-  const requiredLocation = giveaway.requirements.location;
-
-  if (requiredLocation && !participantLocation) // location is required but no location was provided
-    return ParticipantState.PENDING; // participant pending acceptance
-  else if (
-    (requiredLocation &&
-      isValidLocation(participantLocation, requiredLocation)) ||
-    !requiredLocation
-  )
-    // location required and valid or not required
+  const locationId = giveaway.requirements.location;
+
+  if (locationId) {
+    // location is required but no location was provided
+    if (!participantLocation) {
+      return ParticipantState.PENDING; // participant pending acceptance
+    }
+    // location is required and was provided
+    else {
+      const requiredLocation = await Location.findById(locationId);
```

Comment:
No need to make a new request to the database to get the location. Just add `.populate('requirements.location')` to the giveaway request in the `addParticipant` function and it will return the location data in the giveaway